packages/loaders/json-schema/src/scalars.ts

```diff
@@ -28,3 +30,45 @@ export const ObjMapScalar = new GraphQLScalarType({
     return null;
   },
 });
+
+export const GraphQLCustomDateTime = new GraphQLScalarType({
+  name: 'CustomDateTime',
+  description:
+    'Custom DateTime scalar that wraps GraphQLDateTimeConfig and converts non-RFC 3339 strings to RFC 3339.',
+  serialize: value => {
+    // helper to check if the date-time string is RFC 3339 compliant
+    const validateDateTime = (dateTimeString: string) => {
+      dateTimeString =
+        dateTimeString === null || dateTimeString === void 0
+          ? void 0
+          : dateTimeString.toUpperCase();
+      const RFC_3339_REGEX =
+        /^(\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])T([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60))(\.\d{1,})?(([Z])|([+|-]([01][0-9]|2[0-3]):[0-5][0-9]))$/;
+      // Validate the structure of the date-string
+      if (!RFC_3339_REGEX.test(dateTimeString)) {
+        return false;
+      }
+      // Check if it is a correct date using the javascript Date parse() method.
+      const time = Date.parse(dateTimeString);
+      if (time !== time) {
```

Comment:
Should always be false? Superfluous check? Unless I'm missing something.

Reply:
This is actually an intentional check for the value NaN. In JavaScript, NaN (which stands for "Not-a-Number") is a unique value that isn't equal to any other value, including itself. So, the condition (time !== time) is true if and only if time is NaN. This is a standard and recommended way to check for NaN because it's concise and performs better than the global isNaN function in certain edge cases.